### DISCUSSION TITLE
Libraries

### DISCUSSION THREAD

**Discussion by SKZBadHabit** [pschatzmann/pico-arduino#2] [Q&A] [answered]
Eine dumme Frage kann ich eine Arduino Library meinem Code hinzufügen ohne diese zu veränderen?

**Comment by pschatzmann** [comment]
Grundsätzlich schon, nur wird dann CMakeLists.txt ein bisschen komplexer:

- include_directories( ) muss zusätzlich den Pfad zu den .h files in der Arduino Library enhalten
- add_executable() muss zusätzlich alle .c oder .cpp Source Dateien der Library enhalten. 

Diese kann man mit Hilfe von

  file({GLOB | GLOB_RECURSE} <out-var> [...] [<globbing-expr>...])

ziemlich einfach bestimmen und in eine Varible laden.

**Comment by pschatzmann** [comment, marked as answer]
I added an example which automatically downloads an Arduino library from github and uses it:
 https://github.com/pschatzmann/pico-arduino/tree/main/examples/example_with_arduino_library

See my blog at https://www.pschatzmann.ch/home/2021/02/10/using-arduino-libraries-on-the-pico/

**Comment by SKZBadHabit** [comment]
Du bist ein Wahnsinn! Dankeschön

Hab nämlich vor meine Selbstgebaute Ergo Tastatur mit eigener Software auf
einen schnelleren Chip zu portieren und natürlich macht das zeug einfach
Spaß.


Falls es dich interessiert ein paar Bilder zu meiner selbstgelaserten
Tastatur:

https://1drv.ms/u/s!Ahzgzn-Rls9ghuZTZtlwwL6tq8n3GQ?e=Vsap2Y

Am Mi., 10. Feb. 2021 um 11:42 Uhr schrieb Phil Schatzmann <
notifications@github.com>:
